test/commands/push.js

```diff
@@ -0,0 +1,120 @@
+'use strict'
+
+const cli = require('heroku-cli-util')
+const cmd = require('../..').commands.find(c => c.topic === 'container' && c.command === 'push')
+const expect = require('unexpected')
+const sinon = require('sinon')
+
+const Sanbashi = require('../../lib/sanbashi')
+var sandbox
+
+describe('container push', () => {
+  beforeEach(() => {
+    cli.mockConsole()
+    sandbox = sinon.sandbox.create()
+  })
+  afterEach(() => sandbox.restore())
+
+  it('gets a build error', () => {
+    let dockerfiles = sandbox.stub(Sanbashi, 'getDockerfiles')
+      .returns(['/path/to/Dockerfile'])
+    let build = sandbox.stub(Sanbashi, 'buildImage').throws()
+
+    return cmd.run({app: 'testapp', args: ['web'], flags: {}})
+      .then(() => expect(cli.stderr, 'to contain', 'docker build exited with Error: Error'))
+      .then(() => expect(cli.stdout, 'to contain', 'Building web (/path/to/Dockerfile)'))
+      .then(() => sandbox.assert.calledOnce(dockerfiles))
+      .then(() => sandbox.assert.calledOnce(build))
+  })
+
+  it('gets a push error', () => {
+    let dockerfiles = sandbox.stub(Sanbashi, 'getDockerfiles')
+      .returns(['/path/to/Dockerfile'])
+    let build = sandbox.stub(Sanbashi, 'buildImage')
+    let push = sandbox.stub(Sanbashi, 'pushImage').throws()
+
+    return cmd.run({app: 'testapp', args: ['web'], flags: {}})
+      .then(() => expect(cli.stderr, 'to contain', 'docker push exited with Error: Error'))
+      .then(() => expect(cli.stdout, 'to contain', 'Building web (/path/to/Dockerfile)'))
+      .then(() => sandbox.assert.calledOnce(dockerfiles))
+      .then(() => sandbox.assert.calledOnce(build))
+      .then(() => sandbox.assert.calledOnce(push))
+  })
+
+  it('pushes to the docker registry', () => {
+    let dockerfiles = sandbox.stub(Sanbashi, 'getDockerfiles')
+      .returns(['/path/to/Dockerfile'])
+    let build = sandbox.stub(Sanbashi, 'buildImage')
+      .withArgs('/path/to/Dockerfile', 'registry.heroku.com/testapp/web', false, [])
+    let push = sandbox.stub(Sanbashi, 'pushImage')
+      .withArgs('registry.heroku.com/testapp/web', false)
+
+    return cmd.run({app: 'testapp', args: ['web'], flags: {}})
+      .then(() => expect(cli.stderr, 'to be empty'))
+      .then(() => expect(cli.stdout, 'to contain', 'Building web (/path/to/Dockerfile)'))
+      .then(() => expect(cli.stdout, 'to contain', 'Pushing web (/path/to/Dockerfile)'))
+      .then(() => sandbox.assert.calledOnce(dockerfiles))
+      .then(() => sandbox.assert.calledOnce(build))
+      .then(() => sandbox.assert.calledOnce(push))
+  })
+
+  it('pushes the standard dockerfile to non-web', () => {
```

Comment:
This is @cvx's use-case, which makes sense. But I think we should also handle `container:push worker` if you have `Dockerfile.worker`.
We may even be able to get rid of the `--recursive` flag.